components/RecentlyPlayedDrawer.tsx

```diff
@@ -19,318 +15,47 @@ import Animated, {
-      const month = String(easternDate.getMonth() + 1).padStart(2, "0");
-      const day = String(easternDate.getDate()).padStart(2, "0");
-
-      const dateStr = `${year}-${month}-${day}`;
-
-      const songs = await fetchShowPlaylist(currentShow, dateStr);
-      setShowPlaylists([{ showName: currentShow, songs }]);
-
-      // If current show has no songs, mark for auto-load of previous show
-      if (songs.length === 0) {
-        shouldTriggerAutoLoad = true;
-      }
-    } catch (err) {
-      setError(`Failed to load playlist for ${currentShow}`);
-      debugError('Error fetching current show playlist:', err);
-      setShowPlaylists([]);
-    } finally {
-      if (isRefresh) {
-        setRefreshing(false);
-      } else {
-        setLoading(false);
-      }
-
-      // Trigger auto-load after loading state is cleared
-      if (shouldTriggerAutoLoad) {
-        setShouldAutoLoadPrevious(true);
-      }
-    }
-  }, [currentShow, fetchShowPlaylist]);
-
-  const loadPreviousShow = useCallback(async () => {
-    // Determine which show to find the previous show for
-    const lastLoadedShow = showPlaylists.length > 0 ? showPlaylists[showPlaylists.length - 1].showName : currentShow;
-
-    if (!lastLoadedShow || !currentShow || currentShow === 'WMBR 88.1 FM' || loadingMore || hasReachedEndOfDay) {
-      return;
-    }
-
-    setLoadingMore(true);
-
-    try {
-      const scheduleService = ScheduleService.getInstance();
-      const previousShow = await scheduleService.findPreviousShow(lastLoadedShow);
-
-      if (!previousShow) {
-        setHasReachedEndOfDay(true);
-        return;
-      }
-
-      // Check if we've already loaded this specific previous show to prevent duplicates
-      const alreadyLoaded = showPlaylists.some(playlist => playlist.showName === previousShow.show.name);
-      if (alreadyLoaded) {
-        return;
-      }
-
-      try {
-        const songs = await fetchShowPlaylist(previousShow.show.name, previousShow.date);
-
-        setShowPlaylists(prev => [...prev, {
-          showName: previousShow.show.name,
-          songs
-        }]);
-      } catch (playlistError) {
-        // Handle 404 or other playlist fetch errors gracefully - still add the show with empty songs
-        setShowPlaylists(prev => [...prev, {
-          showName: previousShow.show.name,
-          songs: []
-        }]);
-      }
-    } catch (err) {
-      debugError('Error loading previous show:', err);
-    } finally {
-      setLoadingMore(false);
-    }
-  }, [currentShow, showPlaylists, loadingMore, hasReachedEndOfDay, fetchShowPlaylist]);
-
-  // Clear playlist data when current show changes
-  useEffect(() => {
-    setShowPlaylists([]);
-    setHasReachedEndOfDay(false);
-    setShouldAutoLoadPrevious(false);
-    setError(null);
-  }, [currentShow]);
-
-  // Load playlist data when drawer becomes visible and we have a current show
+  // When the screen regains focus, ensure the drawer snaps to the peek position
```

Comment:
thanks for the `useEffect` summary comments! I find it very helpful for when a useEffect's intent isn't clear from a quick read